src/tool/subcommands/archive_cmd.rs

```diff
@@ -373,6 +391,58 @@ async fn do_export(
     Ok(())
 }
 
+// FIXME: Testing with diff snapshots can be significantly improved. Tracking
+// issue: https://github.com/ChainSafe/forest/issues/3347
+/// Merge a set of snapshots (diff snapshots or lite snapshots). The output
+/// snapshot links to the heaviest tipset in the input set.
+async fn merge_snapshots(
+    snapshot_files: Vec<PathBuf>,
+    output_path: PathBuf,
+    force: bool,
+) -> Result<()> {
+    use crate::db::car::forest;
+
+    let store = ManyCar::try_from(snapshot_files)?;
+    let heaviest_tipset = store.heaviest_tipset()?;
+    let roots = Vec::<Cid>::from(&heaviest_tipset.key().cids);
+
+    if !force && output_path.exists() {
+        let have_permission = Confirm::with_theme(&ColorfulTheme::default())
+            .with_prompt(format!(
+                "{} will be overwritten. Continue?",
+                output_path.to_string_lossy()
+            ))
+            .default(false)
+            .interact()
+            // e.g not a tty (or some other error), so haven't got permission.
+            .unwrap_or(false);
+        if !have_permission {
+            return Ok(());
+        }
+    }
+
+    let mut writer = BufWriter::new(tokio::fs::File::create(&output_path).await.context(
+        format!(
+            "unable to create a snapshot - is the output path '{}' correct?",
+            output_path.to_str().unwrap_or_default()
+        ),
+    )?);
+
+    // Stream all available blocks from heaviest_tipset to genesis.
+    let blocks = stream_graph(&store, heaviest_tipset.chain(&store), 0);
```

Comment:
question: Is it the intention here to ignore dead links? Otherwise looks good.

Reply:
Yes, dead links should be ignored.